### PR TITLE
[SPARK-34604][PYTHON][TESTS] Use eventually in TaskContextTestsWithWorkerReuse.test_task_context_correct_with_python_worker_reuse

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

`TaskContextTestsWithWorkerReuse.test_task_context_correct_with_python_worker_reuse` can be flaky and fails sometimes:

```
======================================================================
ERROR [1.798s]: test_task_context_correct_with_python_worker_reuse (pyspark.tests.test_taskcontext.TaskContextTestsWithWorkerReuse)
...
test_task_context_correct_with_python_worker_reuse
    self.assertTrue(pid in worker_pids)
AssertionError: False is not true

----------------------------------------------------------------------
```

I suspect that the Python worker was killed for whatever reason and new attempt created a new Python worker.

This PR fixes the flakiness simply by retrying the test case.

### Why are the changes needed?

To make the tests more robust.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Manually tested it by controlling the conditions manually in the test codes.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @viirya would you mind taking a quick look please?

**Review by viirya — APPROVED** [review]
lgtm

File: python/pyspark/tests/test_taskcontext.py [hunk 1]
```diff
@@ -282,6 +281,11 @@ def context(iterator):
         self.assertTrue(bps == (-1, -1))
         for pid in pids:
             self.assertTrue(pid in worker_pids)
+        return True
+
+    def test_task_context_correct_with_python_worker_reuse(self):
+        eventually(
+            self.check_task_context_correct_with_python_worker_reuse, catch_assertions=True)
```
Comment (viirya) [inline python/pyspark/tests/test_taskcontext.py:288]: nit: we may add few comment like the one in the description, e.g. "Retrying the check as reusing Python workers actually sort of makes the best effort reuse the Python workers."

Comment (HyukjinKwon) [reply python/pyspark/tests/test_taskcontext.py:288]: Sure sure 👌

Comment (viirya) [reply python/pyspark/tests/test_taskcontext.py]: thank you :)

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.1 and branch-3.0.

**Pull request closed without merging**